src/highdicom/frame.py

```diff
@@ -0,0 +1,102 @@
+import logging
+from io import BytesIO
+
+import numpy as np
+from PIL import Image
+from pydicom.dataset import Dataset
+from pydicom.pixel_data_handlers.numpy_handler import pack_bits
+from pydicom.pixel_data_handlers.rle_handler import rle_encode_frame
+from pydicom.uid import (
+    ExplicitVRBigEndian,
+    ExplicitVRLittleEndian,
+    ImplicitVRLittleEndian,
+    JPEGBaseline,
+    JPEG2000Lossless,
+    RLELossless,
+)
+
+
+def encode_frame(
+    array: np.ndarray,
+    metadata: Dataset,
+) -> bytes:
+    """Encodes pixel data of an individual frame.
```

Comment:
Good point! What about adding the following check in `encode_frame()`?

```python
if (self.Rows * self.Columns * self.SamplesPerPixel) % 8 != 0:
    raise ValueError('Size of frame is not a multiple of 8 and can thus not be bit packed')
```

We would then also need to change the way the function is used by `highdicom.seg.sop.Segmentation`.

Reply:
Having spent quite a while carefully implementing this to make sure it does work with arrays that are not a multiple of 8 so that there are not limitations on the segmentations that can be encoded in the bit packed way (and therefore displayable in OHIF etc etc), I'm fairly reluctant to remove this capability without good reason. It would also break backwards compatibility

Reply:
Good point. I think that is a special enough case. I reverted the approach to encoding of segmentation frames for native transfer syntaxes (including bit packed), but use the `frame.encode_frame()` function for encapsulated transfer syntaxes. That should be a good compromise.

Addressed with eba8f2f41992237e699e835ecc473148277a463a

Reply:
Looks good. However now there is a check in `_encode_frame` that the input array is a multiple of 8, which is unnecessarily restrictive. `_encode_frame` is perfectly capable of handling cases that are not a multiple of 8 - it just passes through to pydicom's `pack_bits`, which does the necessary padding and you don't need to worry about it.

The complicated situation is when you have *multiple* frames, each of which have  a number of pixels that does not divide by 8, and you want to bit pack them *together*. However since this function encodes just a single frame, this should not arise if used correctly.